content/webapp/components/IIIFViewer/MainViewer.tsx

```diff
@@ -469,39 +464,62 @@ const MainViewer: FunctionComponent = () => {
-    // Keeping this very simple for now - I think it'll just apply to audio, video and of course mixed works containing either of them.
-    // I would like to also have it return false for PDFs, but they currently do have a thumbnail image.
-    setHasImagesOnly(!canvases?.find(canvas => !canvas.thumbnailImage?.url));
-  }, []);
+  const currentCanvas = canvases?.[queryParamToArrayIndex(canvas)];
+  const displayItems = currentCanvas ? getDisplayItems(currentCanvas) : [];
+  const useFixedSizeList = !hasNonImages(canvases);
+  if (!useFixedSizeList) {
+    setShowFullscreenControl(false);
+  }
 
   return (
-    <div data-testid="main-viewer">
-      <FixedSizeList
-        width={mainAreaWidth}
-        style={{ width: `${mainAreaWidth}px`, margin: '0 auto' }}
-        height={mainAreaHeight}
-        itemCount={canvases?.length || 0}
-        itemData={{
-          scrollVelocity,
-          canvases: canvases || [],
-          setShowZoomed,
-          rotatedImages,
-          errorHandler,
-          externalAccessService,
-          canvas,
-          accessToken,
-          placeholderId,
-        }}
-        // We want images to render bigger than their height for readability
-        // But other things should be centered and therefore based on the height.
-        itemSize={hasImagesOnly ? mainAreaWidth : mainAreaHeight}
-        onItemsRendered={debounceHandleOnItemsRendered.current}
-        onScroll={handleOnScroll}
-        ref={mainViewerRef}
-      >
-        {ItemRenderer}
-      </FixedSizeList>
+    <div data-testid="main-viewer" style={{ height: '100%' }}>
+      {useFixedSizeList ? (
+        <FixedSizeList
+          width={mainAreaWidth}
+          style={{ width: `${mainAreaWidth}px`, margin: '0 auto' }}
+          height={mainAreaHeight}
+          itemCount={canvases?.length || 0}
+          itemData={{
+            scrollVelocity,
+            canvases: canvases || [],
+            setShowZoomed,
+            rotatedImages,
+            errorHandler,
+            externalAccessService,
+            canvas,
+            accessToken,
+            placeholderId,
+          }}
+          itemSize={mainAreaWidth}
+          onItemsRendered={debounceHandleOnItemsRendered.current}
+          onScroll={handleOnScroll}
+          ref={mainViewerRef}
+        >
+          {ItemRenderer}
+        </FixedSizeList>
+      ) : (
+        <>
+          {displayItems.map((item, i) => {
+            return (
+              <>
+                {currentCanvas ? (
+                  <ItemWrapper key={i}>
+                    <IIIFItem
+                      placeholderId={placeholderId}
+                      item={item}
+                      i={1}
+                      canvas={currentCanvas}
+                      titleOverride={`${canvas}/${canvases?.length}`}
+                      exclude={[]}
+                    />
+                  </ItemWrapper>
+                ) : null}
+                <CanvasPaginator />
```

Comment:
For my understanding; the Paginator is redeclared of every item and can be there even if there is no `currentCanvas`? Where could this happen?